dbms/src/DataStreams/SummingSortedBlockInputStream.cpp

```diff
@@ -31,7 +36,20 @@ String SummingSortedBlockInputStream::getID() const
 
 void SummingSortedBlockInputStream::insertCurrentRow(ColumnPlainPtrs & merged_columns)
 {
-    for (size_t i = 0; i < num_columns; ++i)
+    for (auto & desc : columns_to_aggregate)
+    {
+        // Do not insert if the aggregation state hasn't been created
+        if (desc.created)
+        {
+            desc.function->insertResultInto(desc.state.data(), *desc.merged_column);
+            desc.function->destroy(desc.state.data());
```

Comment:
The code is not exception safe: destroy is not called in exception, memory leak when the function with non-trivial state (sumMap) was used. Probably you should also add a descructor for AggregateDescription.

Reply:
Good point! Added try/catch and an explicit state destructor.